From f07832bd99689f9b4fe62722e1423c10ca5450bd Mon Sep 17 00:00:00 2001
From: Serkan Bekdemir <37607-SerkanB@users.noreply.drupalcode.org>
Date: Wed, 12 Jul 2023 13:58:38 +0000
Subject: [PATCH] #3344369 Call form-alter for dropzonejs-form.

---
 focal_point.module | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/focal_point.module b/focal_point.module
index 90a35da..f084033 100644
--- a/focal_point.module
+++ b/focal_point.module
@@ -114,6 +114,15 @@ function focal_point_imce_supported_widgets_alter(array &$widgets) {
   $widgets[] = 'image_focal_point';
 }
 
+/**
+ * Implements hook_form_FORM_ID_alter().
+ */
+function focal_point_form_media_library_add_form_dropzonejs_alter(array &$form, FormStateInterface $form_state) {
+  // Dropzonejs changed their form-id for the media-library-form.
+  // Call our form-alter for the old form-id to make sure focal_point is added.
+  focal_point_form_media_library_add_form_upload_alter($form, $form_state);
+}
+
 /**
  * Implements hook_form_FORM_ID_alter().
  */
-- 
GitLab
